btor/src/boolector_term.cpp

```diff
@@ -267,9 +265,27 @@ Sort BoolectorTerm::get_sort() const
   return sort;
 }
 
+bool BoolectorTerm::is_symbol() const
+{
+  // functions, parameters, and symbolic constants are all symbols
+  auto bkind = bn->kind;
+  return !negated
+         && ((bkind == BTOR_VAR_NODE) || (bkind == BTOR_UF_NODE)
+             || (bkind == BTOR_PARAM_NODE));
+}
+
+bool BoolectorTerm::is_param() const
+{
+  return !negated && (bn->kind == BTOR_PARAM_NODE);
+}
+
 bool BoolectorTerm::is_symbolic_const() const
 {
-  return is_sym;
+  bool is_fun = boolector_is_fun(btor, node);
+  auto bkind = bn->kind;
+  bool is_sym =
+      !negated && ((bkind == BTOR_VAR_NODE) || (bkind == BTOR_UF_NODE));
```

Comment:
are we sure that we want to check for uf-node?

Reply:
Yeah so the way I've set it up is that `is_symbol` returns true for any uninterpreted constant, or function. `is_symbolic_const` only returns true for uninterpreted constants, and `is_param` only returns true for parameters. Does that seem reasonable?

Reply:
Oh I see, yeah this could be simpler. I just copy and pasted. But I could just take out the UF_NODE one and then not have to check if it's a function separately. I'll do that.

Reply:
Made it simpler in e3e74f36278132d722bf3991c6515fa38de16c1b